native/core/src/execution/operators/copy.rs

```diff
@@ -116,10 +116,9 @@ impl ExecutionPlan for CopyExec {
         self: Arc<Self>,
         children: Vec<Arc<dyn ExecutionPlan>>,
     ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
-        let input = Arc::clone(&self.input);
-        let new_input = input.with_new_children(children)?;
+        assert!(children.len() == 1);
         Ok(Arc::new(CopyExec {
-            input: new_input,
+            input: Arc::clone(&children[0]),
```

Comment:
Adding the optimizer highlighted that the previous implementation of `CopyExec::with_new_children` was incorrect.